### PR TITLE
NiceGUI for Visualization Issue #73

### PR DESCRIPTION
**Pull request by ahmetutku**
In this pull request, I have addressed issue #73 (https://github.com/MLDSAI/OpenAdapt/issues/73) by implementing a transformation of the visualize.py file using NiceGUI. The images and screenshots have been converted to videos, which prevents the screen from changing color with hover and enables the user to see the action the info box is referring to. In addition, I have added a title and the company logo to enhance the aesthetic appeal of the file.

Please note that the results can be accessed through the following command: (Please note that the first command is used to ensure that the file.visualize1.py is located in the current puterbot folder).

cd puterbot 
python -m visualize1.py

The next steps that I propose include the implementation of the results of issue #85, the addition of more filters (as shown in the image in the comments above), and the incorporation of a button that shows/hides children's information to improve readability. If given the green light, another proposed next step would be to delete the original visualize.py file and rename visualize1.py to visualize.py.

Thank you for your time and consideration.

### CODE REVIEW

**Comment by 0dm** [comment]
Thank you for tackling that issue! 

Looks good!
image

Would it be possible to draw the keypresses? Like 
image

I suggest we change the MLDSAI here to OpenAdapt, since that's the program:
image

In the future, could you please include the requirements necessary? No big deal, would just save a little time (pipreqs)😅

For now, we shouldn't entirely replace visualize (we can deprecate it in favour for yours), just because there's a significant difference in execution time, here's a test with three screenshots:

`time python3 -m puterbot.visualize`
> 8.20s user 2.09s system 129% cpu 7.946 total

`python3 -m visualize1.pyl`
>  76.38s user 4.13s system 102% cpu 1:18.21 total

Overall, great work!

File: puterbot/strategies/demo.py [hunk 1]
```diff
@@ -6,6 +6,8 @@
     $ python puterbot/replay.py DemoReplayStrategy
 """
 
+from svg import get_text_from_svg
```
Comment (0dm) [inline puterbot/strategies/demo.py:9]: unused

File: puterbot/strategies/naive.py [hunk 2]
```diff
@@ -1,26 +1,23 @@
...
+from puterbot.strategies.ocr_mixin import OCRReplayMixin
 
 
 DISPLAY_EVENTS = False
 REPLAY_EVENTS = True
 SLEEP = True
 
 
-class NaiveReplayStrategy(BaseReplayStrategy):
+class NaiveReplayStrategy(OCRReplayMixin, BaseReplayStrategy):
```
Comment (0dm) [inline puterbot/strategies/naive.py:20]: Why modify NaiveReplay?

File: puterbot/visualize.py [hunk 3]
```diff
@@ -25,7 +25,7 @@
     rows2dicts,
 )
 
-
+#Global variables & Formatting:
```
Comment (0dm) [inline puterbot/visualize.py:28]: use Black for formatting

File: puterbot/visualize1.py [hunk 4]
```diff
@@ -0,0 +1,328 @@
+import html
```
Comment (0dm) [inline puterbot/visualize1.py:1]: many unused imports

File: puterbot/visualize1.py [hunk 5]
```diff
@@ -0,0 +1,328 @@
...
+    return rval
+
+def dict2html(obj, max_children=5):
+   if isinstance(obj, list):
+       children = [dict2html(value) for value in obj]
+       if max_children is not None and len(children) > max_children:
+           all_children = children
+           children = evenly_spaced(children, max_children)
+           children = indicate_missing(children, all_children, "...")
+       html_str = "\n".join([f'<div style="border: 1px solid black; padding: 5px;">{child_html}</div>' if isinstance(child_obj, dict) else child_html for child_obj, child_html in zip(obj, children)])
```
Comment (0dm) [inline puterbot/visualize1.py:145]: line length too long

File: puterbot/visualize.py (hunk 3 above)
Comment (0dm) [inline puterbot/visualize.py:28]: Block comments should have one space before the pound sign (#) and the comment itself.

File: puterbot/visualize1.py [hunk 6]
```diff
@@ -0,0 +1,328 @@
...
+            every_idx += 1
+            skipped = True
+        if skipped:
+            rval.append(indicator)
+        rval.append(some[some_idx])
+        some_idx += 1
+        every_idx += 1
+    return rval
+
+def dict2html(obj, max_children=5):
```
Comment (0dm) [inline puterbot/visualize1.py:138]: indention below

File: puterbot/visualize1.py [hunk 7]
```diff
@@ -0,0 +1,328 @@
...
+           """
+           for key, value in obj.items()
+           if value not in EMPTY
+       ])
+       html_str = f"<table>{rows_html}</table>"
+   else:
+       html_str = html.escape(str(obj))
+   return html_str
+
+def html2table(html):
```
Comment (0dm) [inline puterbot/visualize1.py:162]: please format file with Black

**Review by 0dm — COMMENTED** [review]
some comments regarding style

File: puterbot/strategies/naive.py (hunk 2 above)
Comment (ahmetutku) [reply puterbot/strategies/naive.py:20]: It was from another issue I was working on I thought I reset the changes but apparently not.

**Pull request closed without merging**